src/server/services/dbManager.ts

```diff
@@ -15,12 +15,79 @@ export function deNormalizeUnit(unitId: number) {
 }
 
 export function getAllUnits(): Promise<Unit[]> {
-  return isDBConnected().then((connected: boolean) => {
+  return dbService.isDBConnected().then((connected: boolean) => {
     if (!connected) {
       throw new Error('Database unavailable.');
     }
-    return dbGetAllUnits();
-  }).then((units: Unit[]) => units).catch((err) => {
-    throw new Error('An unexpected error has occurs.');
+    return dbService.getAllUnits();
+  });
+}
+
+/**
+ * Retrieve the unit type of the given unit id
+ * @param {Unit} unitId
+ * @returns {Unit}
+ */
+export function getUnit(unitId: number): Promise<Unit> {
+  return new Promise<Unit>((res) => {
+    res(new Unit(8, 'The 08th MS Team', UnitType.BRIGADE));
+  });
+}
+
+/**
+ * Retrieve the specific parent unit id of the given child unit id
+ * @param {number} unitId
+ * @param {Unit} parentUnitType
+ * @returns {number}
+ */
+export function getParent(unitId: number, parentUnitType: UnitType): Promise<Unit> {
```

Comment:
Should we return only the id instead of the whole unit in these type of functions?

Reply:
That was the original thought I had too, however, the contract in the design specification in the DBManager return Units instead of "number". I am happy to change back to number and have the UX replace them on frontend

Reply:
Maybe we can just change them after everything works.